src/main/kotlin/com/learnspigot/bot/command/ToggleCommand.kt

```diff
@@ -0,0 +1,36 @@
+package com.learnspigot.bot.command
+
+import com.learnspigot.bot.LearnSpigotBot.Companion.replyEmbed
+import net.dv8tion.jda.api.JDA
+import net.dv8tion.jda.api.entities.Guild
+import dev.minn.jda.ktx.events.onCommand
+import dev.minn.jda.ktx.interactions.commands.restrict
+import dev.minn.jda.ktx.interactions.commands.upsertCommand
+
+class ToggleCommand(private val guild: Guild, private val bot: JDA) {
+    fun toggleCommand() {
+        guild.upsertCommand("toggleping", "Toggle whether or not you want to be pinged when you got a reputation") {
+            restrict(guild = true)
+            bot.onCommand("toggleping") {
+                val role = guild.getRoleById(System.getenv("PING_ROLE_ID"))
+                if (it.member!!.roles.contains(role)) {
+                    guild.removeRoleFromMember(it.member!!, role!!).queue()
```

Comment:
Use mentions of user instead of roles. I would add a boolean to their profile data called `mentionRep` or something and toggle that instead

Reply:
mb didnt read the code properly, read the latter half of this review